app/server/src/client/preview/render.ts

```diff
@@ -0,0 +1,46 @@
+import { document, Node } from 'global';
+import dedent from 'ts-dedent';
+import { RenderMainArgs, FetchStoryHtmlType } from './types';
+
+const rootElement = document.getElementById('root');
+
+let fetchStoryHtml: FetchStoryHtmlType;
+
+export default async function renderMain({
+  storyFn,
+  id,
+  selectedKind,
+  selectedStory,
+  showMain,
+  showError,
+  forceRender,
+}: RenderMainArgs) {
+  const params = storyFn();
+
+  const element = await fetchStoryHtml(id, params);
```

Comment:
Should we show a useful error here if the user forgets to set this?

Reply:
Yup that's a great idea. It's entirely unrecoverable.

Reply:
Done